src/middleware/mongooseErrorHandler.js

```diff
@@ -9,14 +9,23 @@ module.exports = (err, req, res, next) => {
   // }
   switch (err.name) {
     case errNames.MONGOOSE_VALIDATION:
-      res.boom.badRequest(errMessages.VALIDATION_FAILED, err[err.name]) // trying to add more info as 'data' (See boom docs)
+      const reasons = extractMongooseMessages(err)
+      res.boom.badRequest(errMessages.VALIDATION_FAILED, { reasons })// add more info as 'data' (See boom docs)
       break
     case errNames.MONGOOSE_CAST:
-      res.boom.badRequest(errMessages.INVALID_ID, err[err.name])
+      res.boom.badRequest(errMessages.INVALID_ID)
       break
 
     // unhandled mongoose error
     default:
       res.boom.badImplementation(errMessages.UNHANDLED_MONGOOSE)
   }
 }
+
+const extractMongooseMessages = (mongooseErr) => {
```

Comment:
Shall we put this in a helper? I thought maybe not as it's specifically for the mongoose error handler